include/nn/swkbd/swkbd.h

```diff
@@ -0,0 +1,175 @@
+#pragma once
+
+#include <nn/types.h>
+
+namespace nn {
+namespace swkbd {
+enum Preset { Default, Password, UserName, DownloadCode, Max_Preset };
+
+enum KeyboardMode {
+    ModeLanguageSet1,
+    ModeNumeric,
+    ModeASCII,
+    ModeLanguageSet1Latin,
+    ModeAlphabet,
+    ModeSimplifiedChinese,
+    ModeTraditionalChinese,
+    ModeKorean,
+    ModeLanguageSet2,
+    ModeMax_KeyboardMode,
+    ModeFull,
+    ModeFullLatin
+};
+
+enum InvalidChar {
+    Space = 1 << 1,
+    AtMark = 1 << 2,
+    Percent = 1 << 3,
+    Slash = 1 << 4,
+    BackSlash = 1 << 5,
+    Numeric = 1 << 6,
+    OutsideOfDownloadCode = 1 << 7,
+    OutsideOfMiiNickName = 1 << 8,
+    Force32 = 1 << 9
+};
+
+enum PasswordMode { Show, Hide, Max_PasswordMode };
+
+enum InputFormMode { OneLine, MultiLine, Separate, Max_InputFormMode };
+
+enum InitialCursorPos { First, Last, Max_InitialCursorPos };
+
+enum TextCheckResult { Success, ShowFailureDialog, ShowConfirmDialog, Max_TextCheckResult };
+
+enum DictionaryLang {
+    Japanese,
+    AmericanEnglish,
+    CanadianFrench,
+    LatinAmericanSpanish,
+    Reserved1,
+    BritishEnglish,
+    French,
+    German,
+    Spanish,
+    Italian,
+    Dutch,
+    Portuguese,
+    Russian,
+    Reserved2,
+    SimplifiedChinesePinyin,
+    TraditionalChineseCangjie,
+    TraditionalChineseSimplifiedCangjie,
+    TraditionalChineseZhuyin,
+    Korean,
+    Max_DictionaryLang
+};
+
+struct DictionaryInfo {
+    uint offset;          // 0x0
+    ushort size;          // 0x4
+    DictionaryLang lang;  // 0x6
+};
+
+// KeyboardMode keyboardMode; // 0x0
```

Comment:
What is this?

Reply:
Probably the leftovers of inspection of `KeyboardConfig` and `ShowKeyboardArg`... no idea, and I can't put them in any useful context, so I just removed it.